pwiz_tools/Skyline/TestPerf/DetectionsPlotTest.cs

```diff
@@ -0,0 +1,228 @@
+                paneHistogram.HandleMouseMoveEvent(graphHistogram.GraphControl, evt);
+                Assert.IsNotNull(paneHistogram.ToolTip, "No tooltip found.");
+                WaitForConditionUI(() => paneHistogram.ToolTip.IsVisible, "Tooltip is not visible.");
+                //verify the tooltip text
+                CollectionAssert.AreEqual(HISTOGRAM_TIP_TEXT, paneHistogram.ToolTip.TipLines);
+                
+                paneHistogram.HandleMouseMoveEvent(graphHistogram.GraphControl, new MouseEventArgs(MouseButtons.None, 0, 0, 0, 0));
+                Assert.IsFalse(paneHistogram.ToolTip.IsVisible);
+            });
+            RunUI(() =>
+            {
+                graph.Close();
+                graphHistogram.Close();
+            });
+            WaitForGraphs();
+            Trace.WriteLine(this.GetType().Name + ": Test complete.");
+        }
+
+        private void AssertDataCorrect(DetectionsPlotPane pane, int refIndex, float qValue, bool record = false)
+        {
+            DetectionPlotData data = null;
+            Trace.WriteLine(this.GetType().Name + $": Waiting for data for qValue {qValue} .");
+            WaitForConditionUI(() => (data = pane.CurrentData) != null 
+                                           && pane.CurrentData.QValueCutoff == qValue
+                                           && DetectionPlotData.DataCache.Status == DetectionPlotData.DetectionDataCache.CacheStatus.idle,
+                () => $"Retrieving data for qValue {qValue}, refIndex {refIndex} took too long.");
+            WaitForGraphs();
+            Assert.IsTrue(data.IsValid);
+
+            if (record)
+            {
+                Console.WriteLine(@"Peptides");
+                pane.CurrentData.GetTargetData(DetectionsGraphController.TargetType.PEPTIDE).TargetsCount
+                    .ForEach((cnt) => { Debug.Write($"{cnt}, "); });
+                Console.WriteLine(@"\nPrecursors");
+                pane.CurrentData.GetTargetData(DetectionsGraphController.TargetType.PRECURSOR).TargetsCount
+                    .ForEach((cnt) => { Debug.Write($"{cnt}, "); });
+            }
+
+            Assert.IsTrue(
+                REF_DATA[refIndex].SequenceEqual(
+                    pane.CurrentData.GetTargetData(DetectionsGraphController.TargetType.PEPTIDE).TargetsCount));
+            Assert.IsTrue(
+                REF_DATA[refIndex + 1].SequenceEqual(
+                    pane.CurrentData.GetTargetData(DetectionsGraphController.TargetType.PRECURSOR).TargetsCount));
+        }
+
+        private static MouseEventArgs FindBarPoint(GraphPane pane, Control graph, int index)
+        {
+            using (var g = graph.CreateGraphics())
+            {
+                var iterationCount = 0;
+                var randomInt = new Random();
+                var evt = new MouseEventArgs(MouseButtons.None, 0, 0, 0, 0);
+                while (!(pane.FindNearestObject(evt.Location, g, out var nearestObject, out var objectIndex) &&

```

Comment:
When I run this test under TestRunner (which sets Program.SkylineOffscreen to true), I always get stuck on this line.

Reply:
This is very strange. I can the test both on and off screen without any problems. May be I should explicitly seed the random number generator to make it reproducible.

Reply:
Does it happen only in offscreen mode?